compiler/semstmts.nim

```diff
@@ -731,16 +743,21 @@ proc semFor(c: PContext, n: PNode): PNode =
 proc semRaise(c: PContext, n: PNode): PNode =
   result = n
   checkSonsLen(n, 1)
-  if n.sons[0].kind != nkEmpty:
-    n.sons[0] = semExprWithType(c, n.sons[0])
-    var typ = n.sons[0].typ
-    if typ.kind != tyRef or typ.lastSon.kind != tyObject:
-      localError(n.info, errExprCannotBeRaised)
-
-    # check if the given object inherits from Exception
-    if not typ.lastSon.isException():
-        localError(n.info, "raised object of type $1 does not inherit from Exception",
-                           [typeToString(typ)])
+  if n[0].kind == nkEmpty:
+    return
```

Comment:
not a fan of early returns.

Reply:
done, but now it is many nested if statements which is not good either